### ISSUE TITLE
Unable to detect sensors on 4.18

### ISSUE THREAD

**Issue by JP-Ellis** [dgraziotin/mbpfan#162]
I've noticed recently that mbpfan doesn't work as it should.  It is trying to use legacy sensor paths despite running Linux 4.18.  I'm on Arch Linux and I've only noticed this issue the past 2-3 days (not really sure what triggered it in the first place).

Here's a couple of outputs:

```
 ❯ sudo mbpfan -f -v
Writing a new .pid file with value 1762 at: /var/run/mbpfan.pid
mbpfan[1762]: Writing a new .pid file with value 1762 at: /var/run/mbpfan.pid
Successfully written a new .pid file with value 1762 at: /var/run/mbpfan.pid
mbpfan[1762]: Successfully written a new .pid file with value 1762 at: /var/run/mbpfan.pid
Using legacy sensor path for kernel < 3.15.0
Found 0 sensors
mbpfan[1762]: mbpfan could not detect any temp sensor. Please contact the developer.
mbpfan could not detect any temp sensor. Please contact the developer.
```

```
 ❯ uname -r
4.18.3-arch1-1-ARCH
```

Let me know if you want me to test other things, or want other outputs from my computer.

**Comment by gaul** [comment]
Could you attach the output of: `find /sys/devices/platform/coretemp.0`?  Which version of `mbpfan` do you use?  Master changes some of this logic and I worry it has a regression.

**Comment by dixflat** [comment]
I have the same problem on a a Macbook Air mid 2012 running Fedora 28 (4.17.14-202.fc28.x86_64).

Here is the **find /sys/devices/platform/coretemp.0** output in my case:
/sys/devices/platform/coretemp.0
/sys/devices/platform/coretemp.0/uevent
/sys/devices/platform/coretemp.0/power
/sys/devices/platform/coretemp.0/power/runtime_active_time
/sys/devices/platform/coretemp.0/power/runtime_status
/sys/devices/platform/coretemp.0/power/autosuspend_delay_ms
/sys/devices/platform/coretemp.0/power/runtime_suspended_time
/sys/devices/platform/coretemp.0/power/control
/sys/devices/platform/coretemp.0/driver
/sys/devices/platform/coretemp.0/driver_override
/sys/devices/platform/coretemp.0/subsystem
/sys/devices/platform/coretemp.0/hwmon
/sys/devices/platform/coretemp.0/hwmon/hwmon0
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp3_crit
/sys/devices/platform/coretemp.0/hwmon/hwmon0/uevent
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp2_crit_alarm
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp2_crit
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp2_max
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp3_input
/sys/devices/platform/coretemp.0/hwmon/hwmon0/power
/sys/devices/platform/coretemp.0/hwmon/hwmon0/power/runtime_active_time
/sys/devices/platform/coretemp.0/hwmon/hwmon0/power/runtime_status
/sys/devices/platform/coretemp.0/hwmon/hwmon0/power/autosuspend_delay_ms
/sys/devices/platform/coretemp.0/hwmon/hwmon0/power/runtime_suspended_time
/sys/devices/platform/coretemp.0/hwmon/hwmon0/power/control
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp1_label
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp1_crit
/sys/devices/platform/coretemp.0/hwmon/hwmon0/device
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp3_max
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp3_crit_alarm
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp2_label
/sys/devices/platform/coretemp.0/hwmon/hwmon0/subsystem
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp1_input
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp1_crit_alarm
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp3_label
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp1_max
/sys/devices/platform/coretemp.0/hwmon/hwmon0/name
/sys/devices/platform/coretemp.0/hwmon/hwmon0/temp2_input
/sys/devices/platform/coretemp.0/modalias


MBPFAN output:

Writing a new .pid file with value 7815 at: /var/run/mbpfan.pid
mbpfan[7815]: Writing a new .pid file with value 7815 at: /var/run/mbpfan.pid
Successfully written a new .pid file with value 7815 at: /var/run/mbpfan.pid
mbpfan[7815]: Successfully written a new .pid file with value 7815 at: /var/run/mbpfan.pid
Using legacy sensor path for kernel < 3.15.0
Found 0 sensors
mbpfan[7815]: mbpfan could not detect any temp sensor. Please contact the developer.
mbpfan could not detect any temp sensor. Please contact the developer.

**Comment by gaul** [comment]
Could you confirm that this works in 2.0.2 and fails in master?

**Comment by gaul** [comment]
I tested the fix on Linux 3.13 and 4.4.  Please reopen or comment if symptoms persist.
